AlphaWallet/AppCoordinator.swift

```diff
@@ -268,50 +236,97 @@ class AppCoordinator: NSObject, Coordinator {
         WalletCoordinator(config: config, keystore: keystore, analyticsCoordinator: analyticsService).createInitialWalletIfMissing()
     }
 
+    private func showTransactionsIfNeeded(instantiateRequired: Bool = false) {
+        if inCoordinator != nil {
+            //no-op
+        } else if let pendingCoordinator = pendingInCoordinator {
+            addCoordinator(pendingCoordinator)
+            pendingCoordinator.showTabBar(animated: false)
+
+            pendingInCoordinator = .none
+        } else {
+            //NOTE: or wait until presented
+            guard instantiateRequired else { return }
+            showTransactions(for: keystore.currentWallet, animated: false)
+        }
+    }
+
+    /// Return true if handled
     @discardableResult func handleUniversalLink(url: URL) -> Bool {
         createInitialWalletIfMissing()
-        let inCoordinator = showTransactionsIfNeeded()
 
-        guard let server = RPCServer(withMagicLink: url) else { return false }
+        showTransactionsIfNeeded()
 
-        if config.enabledServers.contains(server) {
-            let universalLinkCoordinator = UniversalLinkCoordinator(
-                analyticsCoordinator: analyticsService,
-                wallet: keystore.currentWallet,
-                config: config,
-                ethPrice: inCoordinator.nativeCryptoCurrencyPrices[server],
-                ethBalance: inCoordinator.nativeCryptoCurrencyBalances[server],
-                tokensDatastore: inCoordinator.tokensStorages[server],
-                assetDefinitionStore: assetDefinitionStore,
-                url: url,
-                server: server
-            )
-
-            universalLinkCoordinator.delegate = self
-            universalLinkCoordinator.start()
+        let handled = universalLinkCoordinator.handleUniversalLinkOpen(url: url)
+        if handled {
+            return true
+        }
+//        //TODO clean up handling of custom URL schemes:
+//        if url.scheme == "wc", let wcUrl = AlphaWallet.WalletConnect.ConnectionUrl(url.absoluteString), let inCoordinator = inCoordinator {
+//            inCoordinator.openWalletConnectSession(url: wcUrl)
+//            return true
+//        }
 
-            let handled = universalLinkCoordinator.handleUniversalLink()
+        //let shouldBeHandledByEip681UrlResolver = Eip681UrlResolver.canHandleOpen(url: url) && inCoordinator != nil
+        //NOTE: avoid displaying error from `assetDefinitionStoreCoordinator.handleOpen(url` while handling eip681 url
+        if let assetDefinitionStoreCoordinator = assetDefinitionStoreCoordinator/*, !shouldBeHandledByEip681UrlResolver*/ {
+            let handled = assetDefinitionStoreCoordinator.handleOpen(url: url)
```

Comment:
```
assetDefinitionStoreCoordinator.handleOpen(url: url)
```

Not sure if this ^ should be called here anymore. It should only be called (or indirectly called) from `AppDelegate.application(_: UIApplication,open:,options:)`, when triggered when we AirDrop an TokenScript/XML/CSS/HTML file. It was less clear prior to the PR because there was a mixing up of URL handling code :) *Maybe* this could be an additional enum case in `MagicLinkUrl` (after renaming `MagicLinkUrl`)? Maybe that's what `.maybeFileUrl` is for?

Reply:
i tried import tokenscript files, works for me. But yep, could be better to add case maybeFileUrl i guess